src/completers/completeState/index.js

```diff
@@ -1,22 +1,44 @@
-import { isStringArray, isValidObject } from '../../utils/typeUtils';
+import { isStringArray, isValidObject, isObjectArray } from '../../utils/typeUtils';
 
-// Given a defaultState, it populates that state with ${key}Loading and ${key}Error
-function completeState(defaultState, ignoredTargets = []) {
-  if (!isValidObject(defaultState)) {
-    throw new Error('Expected an object as a state to complete');
+const isValidCustomCompleter = (targets, completers) =>
+  isStringArray(targets) && isValidObject(completers);
+
+export const customCompleter = (target, completers = {}) =>
+  Object.keys(completers)
+    .map(completer => ({ [`${target}${completer}`]: completers[completer] }))
+    .reduce((acc, value) => ({ ...acc, ...value }), {});
+
+function completeState({ description, targetCompleters = [], ignoredTargets = [] }) {
```

Comment:
I don't think so, for me, that should go in the description attribute. besides you could use a simple object instead.

Reply:
If you put it in the `description`, it will have `loading` and `error` and maybe the user doesn't want that.

I gave a bad example before 😅. Maybe I want
`{a: 1, b: 2, c: 3, cLoading: false, cError: null}`
and I can do 
```js
completeState({
    ignoredTargets: {a: 1, b: 2},
    description: {c: 3}
})
```